### PR TITLE
Replace sGet, sSave macros - part 2 - DOS_FCB, DOS_DTA, DOS_SDA classes

### PR DESCRIPTION
**Pull request by dreamer**
Continuation of #558 and #550.

### CODE REVIEW

File: include/dos_inc.h [hunk 1]
```diff
@@ -560,29 +574,45 @@ class DOS_DTA:public MemStruct{
...
-	Bit8u GetDrive(void);
-	bool Extended(void);
-	void GetAttr(Bit8u & attr);
-	void SetAttr(Bit8u attr);
+
+	void SetSizeDateTime(uint32_t size, uint16_t date, uint16_t time);
+	void GetSizeDateTime(uint32_t &size, uint16_t &date, uint16_t &time) const;
+
+	void FileOpen(uint8_t fhandle);
+	void FileClose(uint8_t &fhandle);
```
[the remark this answers is not in the document]
Comment (dreamer) [reply include/dos_inc.h:594]: Pedantically, almost all functions in these classes wrapping DOS structures are const (because they do not modify memory pointed by `this` - they modify memory pointed bl DOSBox "pointer" `pt`.

As a matter of keeping the interface sane (in case we'll want to refactor these classes in the future), I decided to add `const` only to getters / methods utilizing `SGET_*` macros. Both `FileOpen` and `FileClose` use `SSET_*` macros.

In other words - in these classes I decided to make `const` synonymous to "this function has no side effects", otherwise code like this would be valid, and it might cause problems down the road:
```
const DOS_SomeStruct ds;
ds.SetField(value);
```

**Comment by dreamer** [comment]
Thank for the review, @kcgen :) Off to the next one, which will remove all remaining UBs.
